packages/plugin-core/src/test/suite-integ/VaultAddCommand.test.ts

```diff
@@ -428,3 +430,160 @@ suite("VaultAddCommand", function () {
     });
   });
 });
+
+function enableSelfCOntainedVaults(config: IntermediateDendronConfig) {
+  config.dev!.enableSelfContainedVaults = true;
+  return config;
+}
+describe("GIVEN VaultAddCommand with self contained vaults enabled", function () {
+  const ctx = setupBeforeAfter(this, {});
+
+  describeSingleWS(
+    "WHEN creating and adding a local vault",
+    {
+      ctx,
+      modConfigCb: enableSelfCOntainedVaults,
+    },
+    () => {
+      const vaultName = "my-self-contained-vault";
+      before(async () => {
+        sinon.stub(VSCodeUtils, "showQuickPick").resolves({ label: "local" });
+        sinon.stub(VSCodeUtils, "showInputBox").resolves(vaultName);
+        sinon.stub(vscode.commands, "executeCommand").resolves({}); // stub reload window
+
+        await new VaultAddCommand().run();
+      });
+
+      test("THEN the vault is under `dependencies/localhost`, and is self contained", async () => {
+        const { wsRoot } = ExtensionProvider.getDWorkspace();
+        const vaultPath = path.join(
+          wsRoot,
+          FOLDERS.DEPENDENCIES,
+          FOLDERS.LOCAL_DEPENDENCY,
+          vaultName
+        );
+        expect(await fs.pathExists(vaultPath)).toBeTruthy();
+        expect(
+          await readYAMLAsync(
+            path.join(vaultPath, CONSTANTS.DENDRON_CONFIG_FILE)
+          )
+        ).toBeTruthy();
+      });
+
+      test("THEN the vault was added to the workspace config correctly", async () => {
+        const { wsRoot } = ExtensionProvider.getDWorkspace();
+        const config = DConfig.getOrCreate(wsRoot);
+        const vault = VaultUtils.getVaultByName({
+          vaults: ConfigUtils.getVaults(config),
+          vname: vaultName,
+        });
+        expect(vault?.selfContained).toBeTruthy();
+        expect(vault?.name).toEqual(vaultName);
+        expect(vault?.fsPath).toEqual(
+          path.join(FOLDERS.DEPENDENCIES, FOLDERS.LOCAL_DEPENDENCY, vaultName)
+        );
+      });
+
+      test("THEN the notes in this vault are accessible", async () => {
+        // Since we mock the reload window, need to reload index here to pick up the notes in the new vault
+        await new ReloadIndexCommand().run();
+        const { engine, vaults } = ExtensionProvider.getDWorkspace();
+        const vault = VaultUtils.getVaultByName({
+          vaults,
+          vname: vaultName,
+        });
+        expect(vault).toBeTruthy();
+        const note = NoteUtils.getNoteByFnameFromEngine({
+          fname: "root",
+          vault: vault!,
+          engine,
+        });
+        expect(note).toBeTruthy();
+        expect(note?.vault.name).toEqual(vaultName);
+      });
+    }
+  );
+
+  describeSingleWS(
```

Comment:
we should have a test case for adding a self contained vault that has other vaults in it

Reply:
This is not supported yet, I was planning to tackle that in a next step.